### PR TITLE
nixos/lighttpd: add new mod_authn_* modules

### PR DESCRIPTION
**Pull request by brechtcs**
###### Motivation for this change
Since the last update to this configuration, `lighttpd` has add three new authentication modules. This PR adds support for these.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by brechtcs** [comment]
> (Module updates) Added a release notes entry if the change is significant

Not quite sure if this change would be considered significant enough for the release notes. If so, I can still add one before this is merged.

File: pkgs/servers/http/lighttpd/default.nix [hunk 1]
```diff
@@ -69,6 +69,6 @@ stdenv.mkDerivation rec {
     homepage = "http://www.lighttpd.net/";
     license = lib.licenses.bsd3;
     platforms = platforms.linux ++ platforms.darwin;
-    maintainers = [ maintainers.bjornfor ];
+    maintainers = [ maintainers.bjornfor maintainers.brecht ];
```
Comment (SuperSandro2000) [inline pkgs/servers/http/lighttpd/default.nix:72]:
```suggestion
    maintainers = with maintainers; [ bjornfor brecht ];
```

Comment (brechtcs) [reply pkgs/servers/http/lighttpd/default.nix]: Done

**Comment by brechtcs** [comment]
Discovered some issues with this after some extra testing. Going to close this for now, and open a new PR when it's fixed.

**Comment by brechtcs** [comment]
Superceded by https://github.com/NixOS/nixpkgs/pull/133658

**Pull request closed without merging**